File: Frontend/src/api/lookupsService.ts
import { axiosClient } from './axiosClient';

export interface LookupItem {
  id: number;
  name: string;
}

export const lookupsService = {
  getCompanies: async (): Promise<LookupItem[]> => {
    const res = await axiosClient.get<any[]>('/Lookups/companies');
    return (res.data || []).map((item) => ({
      id: item.id,
      name: item.companyName || item.name || '',
    }));
  },

  getRoles: async (): Promise<LookupItem[]> => {
    const res = await axiosClient.get<any[]>('/Lookups/roles');
    return (res.data || []).map((item) => ({
      id: item.id,
      name: item.roleName || item.name || '',
    }));
  },

  getColleges: async (): Promise<LookupItem[]> => {
    const res = await axiosClient.get<any[]>('/Lookups/colleges');
    return (res.data || []).map((item) => ({
      id: item.id,
      name: item.collegeName || item.name || '',
    }));
  },
};

File: Backend/Services/Implementations/LookupsService.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using summer_training_app.Data;
using summer_training_app.DTOs.Core;
using summer_training_app.Services.Interfaces;

namespace summer_training_app.Services.Implementations
{
    public class LookupsService : ILookupsService
    {
        private readonly SummerTrainingDBContext _context;

        public LookupsService(SummerTrainingDBContext context)
        {
            _context = context;
        }

        public async Task<List<CompanyLookupDto>> GetCompaniesAsync()
        {
            return await _context.Companies
                .Where(c => c.IsApproved && !c.IsDeleted)
                .Select(c => new CompanyLookupDto
                {
                    Id = c.Id,
                    CompanyName = c.CompanyName,
                })
                .ToListAsync();
        }

        public async Task<List<RoleLookupDto>> GetRolesAsync()
        {
            return await _context.Roles
                .Where(r => r.RoleName != "SuperAdmin")
                .Select(r => new RoleLookupDto
                {
                    Id = r.Id,
                    RoleName = r.RoleName
                })
                .ToListAsync();
        }

        public async Task<List<CollegeLookupDto>> GetCollegesAsync()
        {
            return await _context.Colleges
                .Where(c => !c.IsDeleted)
                .Select(c => new CollegeLookupDto
                {
                    Id = c.Id,
                    CollegeName = c.CollegeName,
                })
                .ToListAsync();
        }
    }
}
